src/node_crypto.cc

```diff
@@ -530,46 +530,41 @@ int SSL_CTX_get_issuer(SSL_CTX* ctx, X509* cert, X509** issuer) {
 }
 
 
-// Read a file that contains our certificate in "PEM" format,
-// possibly followed by a sequence of CA certificates that should be
-// sent to the peer in the Certificate message.
-//
-// Taken from OpenSSL - editted for style.
 int SSL_CTX_use_certificate_chain(SSL_CTX* ctx,
-                                  BIO* in,
+                                  X509* x,
+                                  STACK_OF(X509)* extra_certs,
                                   X509** cert,
                                   X509** issuer) {
-  int ret = 0;
-  X509* x = nullptr;
-
-  x = PEM_read_bio_X509_AUX(in, nullptr, CryptoPemCallback, nullptr);
-
-  if (x == nullptr) {
-    SSLerr(SSL_F_SSL_CTX_USE_CERTIFICATE_CHAIN_FILE, ERR_R_PEM_LIB);
-    goto end;
+  if (*issuer != nullptr) {
+    X509_free(*issuer);
+    *issuer = nullptr;
+  }
+  if (*cert != nullptr) {
+    X509_free(*cert);
+    *cert = nullptr;
```

Comment:
Not a real objection but I'm not wildly in love with the implicit freeing of the arguments.  I would have made it a responsibility of the caller and simply CHECK that they're nullptr here.